recipe/meta.yaml

```diff
@@ -9,13 +18,13 @@ source:
   url: https://github.com/quantumlib/qsim/archive/refs/tags/v{{ version }}.tar.gz
   sha256: 81101fe9e6356e54125f703956e787aa366717c1e6e77aea794879b53c84ff60
   patches:
-    - patches/0001-Fix-no-instance-of-constructor-qsim-StateSpaceCUDA-F.patch  # [cuda_compiler_version != "None"]
-    - patches/0002-Remove-PUBLIC-from-target_link_libraries-with-OpenMP.patch  # [cuda_compiler_version != "None"]
-    - patches/0003-Set-pybind11_INCLUDE_DIRS-correctly-for-CUDA.patch          # [cuda_compiler_version != "None"]
+    - patches/0001-Fix-no-instance-of-constructor-qsim-StateSpaceCUDA-F.patch  # [cuda_compiler != "None"]
+    - patches/0002-Remove-PUBLIC-from-target_link_libraries-with-OpenMP.patch  # [cuda_compiler != "None"]
+    - patches/0003-Set-pybind11_INCLUDE_DIRS-correctly-for-CUDA.patch          # [cuda_compiler != "None"]
     - patches/0004-Set-PYTHON_INCLUDE_DIR-in-setup.py.patch
     - patches/0005-Allow-installing-for-Python-3.12.patch
-    - patches/0006-Set-CUDA_ARCHITECTURES-all-cmake-policy-CMP0104.patch       # [cuda_compiler_version != "None"]
-    - patches/0007-Always-skip-test_cirq_qsim_gpu_amplitudes-because-no.patch  # [cuda_compiler_version != "None"]
+    - patches/0006-Set-CUDA_ARCHITECTURES-all-cmake-policy-CMP0104.patch       # [cuda_compiler != "None"]
```

Comment:
I tested it locally that the patch can be removed without introducing the CMake warning below, if `CUDAARCHS` is set:
```
  -- Found pybind11 v2.11.1: /home/conda/feedstock_root/build_artifacts/qsimcirq_1702921273289/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/include;/home/conda/feedstock_root/build_artifacts/qsimcirq_1702921273289/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/include/python3.11
  -- Configuring done (0.5s)
  CMake Warning (dev) in pybind_interface/decide/CMakeLists.txt:
    Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
    empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
    for policy details.  Use the cmake_policy command to set the policy and
    suppress this warning.

    CUDA_ARCHITECTURES is empty for target "qsim_decide".
  This warning is for project developers.  Use -Wno-dev to suppress it.

  CMake Warning (dev) in pybind_interface/cuda/CMakeLists.txt:
    Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
    empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
    for policy details.  Use the cmake_policy command to set the policy and
    suppress this warning.

    CUDA_ARCHITECTURES is empty for target "qsim_cuda".
  This warning is for project developers.  Use -Wno-dev to suppress it.

  CMake Warning (dev) in pybind_interface/custatevec/CMakeLists.txt:
    Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
    empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
    for policy details.  Use the cmake_policy command to set the policy and
    suppress this warning.

    CUDA_ARCHITECTURES is empty for target "qsim_custatevec".
  This warning is for project developers.  Use -Wno-dev to suppress it.

  -- Generating done (0.0s)
```

Reply:
Yes that seems good!

I had no idea which architectures we should build for, so that's why I just selected all. However, I am sure you have the right knowledge here ;)

Reply:
Sounds good, done. 

For NV GPU archs, I usually go to Wiki, the table "Compute Capability, GPU semiconductors and Nvidia GPU board products" is surprisingly useful: https://en.wikipedia.org/wiki/CUDA#GPUs_supported. You can read off the compute capabilities for non-edge devices (i.e., not Tegra/Jetson).

For CMake, it has a way to handle the archs: https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_ARCHITECTURES.html If a number is given, then it means both PTX and SASS for that compute capability would be generated. So, for the `CUDAARCHS` set here (which is an env var that CMake uses to initialize `CMAKE_CUDA_ARCHITECTURES`), it means we want
- Volta SASS (70-real)
- Turing SASS (75-real)
- Ampere SASS (80-real & 86-real)
- Ada SASS (89-real)
- Hopper PTX & SASS (90)

Reply:
I looked up my notes (it's been a while since I touched qsim), I think your patch can be reused. It's just that I am no longer certain if `CUDAARCHS` could actually overwrite the default value set in the patch, so I instead updated the patch to ensure it reads the values in `CUDAARCHS`.